api/routes/auth.js

```diff
@@ -0,0 +1,111 @@
+const { SECRET_KEY } = process.env
+const router = require('express').Router()
+const User = require('../models/user')
+const bcrypt = require('bcrypt')
+const jwt = require('jsonwebtoken')
+
+router.post('/signup', async (req, res, next) => {
+  const status = 201
+  const { username, password, admin } = req.body
+  try {
+    if(!username) {
+      return next({status: 400, message: 'Username was not provided.'})
+    } 
+    if(!password) {
+      return next({status: 400, message: 'Password was not provided.'})
+    } 
+    if(password.length < 8) {
+      return next({status: 400, message: 'Password is less than 8 characters.'})
+    }
+
+    // check to see if a user already exists with that name
+    const user = await User.findOne({ username })
+    // return an error if the user already exists
+    if(user !== null) {
+      return next({status: 400, message: 'Username is already taken.'})
+    } 
+
+    // if the user does not exist, hash the given password
+    const hash = await bcrypt.hash(password, 12)
+    // create a new user with the hashed password
+    const response = await User.create({ username: username, password: hash })
+    const payload = { id: response._id, username: response.username, admin: response.admin }
```

Comment:
Destructuring the response above can be an easy way to make this code a bit more readable.